### PR TITLE
fix: compatible with keadm init/upgrade -- profie

### PR DESCRIPTION
**Pull request by luomengY**
**What type of PR is this?**

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes https://github.com/kubeedge/kubeedge/issues/5601

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Review by Shelley-BaoYue — COMMENTED** [review]
CI `Keadm e2e test` always fail, need to find it out.

**Comment by luomengY** [comment]
> CI `Keadm e2e test` always fail, need to find it out.

We need to merge PR https://github.com/kubeedge/kubeedge/pull/5632 here first

**Comment by Shelley-BaoYue** [comment]
Sorry, I cannot understand this pr and #5632.  #5632 uses kubeedge-version to repleace profile version in CI because we will use --kubeedge-version in future release. But for compatibility, we consider continuing to support --profile in the next few releases, so we have this pr.  So this PR should not depend on #5632 and it should run successfully.

**Comment by Shelley-BaoYue** [comment]
> Sorry, I cannot understand this pr and #5632. #5632 uses kubeedge-version to repleace profile version in CI because we will use --kubeedge-version in future release. But for compatibility, we consider continuing to support --profile in the next few releases, so we have this pr. So this PR should not depend on #5632 and it should run successfully.

https://github.com/kubeedge/kubeedge/pull/5461 causes CI fail, so this PR need to wait for #5632 merged first

**Comment by luomengY** [comment]
> Sorry, I cannot understand this pr and #5632. #5632 uses kubeedge-version to repleace profile version in CI because we will use --kubeedge-version in future release. But for compatibility, we consider continuing to support --profile in the next few releases, so we have this pr. So this PR should not depend on #5632 and it should run successfully.

In 1.17, we actually changed the `--profile version=$version` in e2e to `--profile version`. At that time, we did not notice this change in order to pass e2e. According to the previous logic, we directly used the `version.yaml` in the profile directory of the chart package, but this did not actually specify a version. When compiling, keadm e2e used the default version 1.15.0. Therefore, after compatibility is made, we need to first change the `--profile version` in the keadm e2e test to `--kubedge-version $version`. The PR modification is shown in: https://github.com/kubeedge/kubeedge/pull/5632.

File: keadm/cmd/keadm/app/cmd/cloud/upgrade.go [hunk 1]
```diff
@@ -31,6 +35,18 @@ func NewCloudUpgrade() *cobra.Command {
 		Long: "Upgrade the cloud components to the desired version, " +
 			"it uses helm to upgrade the installed release of cloudcore chart, which includes all the cloud components",
 		RunE: func(cmd *cobra.Command, args []string) error {
+			if opts.Profile != "" && opts.KubeEdgeVersion == "" {
```
[the remark this answers is not in the document]
Comment (luomengY) [reply keadm/cmd/keadm/app/cmd/cloud/upgrade.go]: ok

File: keadm/cmd/keadm/app/cmd/helm/cloudcore.go [hunk 2]
```diff
@@ -187,6 +209,29 @@ func (c *CloudCoreHelmTool) Install(opts *types.InitOptions) error {
 
 // Upgrade uses helm client to upgrade cloudcore release
 func (c *CloudCoreHelmTool) Upgrade(opts *types.CloudUpgradeOptions) error {
+	externValueFile := ""
+	if opts.Profile != "" {
+		// TODO:the version specified through profile will be obsolete.
+		kvs := strings.Split(opts.Profile, "=")
+		if len(strings.Split(opts.Profile, "=")) == 2 {
```
Comment (Shelley-BaoYue) [inline keadm/cmd/keadm/app/cmd/helm/cloudcore.go:216]:
```suggestion
		if len(kvs) == 2 {
```

Comment (luomengY) [reply keadm/cmd/keadm/app/cmd/helm/cloudcore.go]: done

File: keadm/cmd/keadm/app/cmd/helm/cloudcore.go [hunk 3]
```diff
@@ -90,6 +90,29 @@ func NewCloudCoreHelmTool(kubeConfig, kubeedgeVersion string) *CloudCoreHelmTool
 
 // Install uses helm client to install cloudcore release
 func (c *CloudCoreHelmTool) Install(opts *types.InitOptions) error {
+	externValueFile := ""
+	if opts.Profile != "" {
+		// TODO:the version specified through profile will be obsolete.
+		kvs := strings.Split(opts.Profile, "=")
+		if len(strings.Split(opts.Profile, "=")) == 2 {
```
[the remark this answers is not in the document]
Comment (luomengY) [reply keadm/cmd/keadm/app/cmd/helm/cloudcore.go]: done

**Review by Shelley-BaoYue — COMMENTED** [review]
/lgtm 
cc @WillardHu @fisherxu

**Review by fisherxu — COMMENTED** [review]
/approve

**Comment by Shelley-BaoYue** [comment]
please cherry-pick it to release 1.17 : )

**Pull request merged**